core/api/service/chain/impl/chain_api_impl.cpp

```diff
@@ -40,13 +25,17 @@ namespace kagome::api {
     return last_finalized.block_hash;
   }
   outcome::result<common::Hash256> ChainApiImpl::getBlockHash(
-      uint32_t value) const {
+      BlockNumber value) const {
     return block_repo_->getHashByNumber(value);
   }
 
   outcome::result<BlockHash> ChainApiImpl::getBlockHash(
       std::string_view value) const {
-    OUTCOME_TRY(number, unhexNumber(value));
+    // despite w3f specification says, that request contains 32-bit
+    // unsigned integer, we are free to decode more capacious number,
+    // since BlockNumber, which is really being requested
+    // is defined as uint64_t
+    OUTCOME_TRY(number, common::unhexNumber<BlockNumber>(value));
```

Comment:
```suggestion
    OUTCOME_TRY(number, common::unhexNumber(value));
```
I guess C++17 can handle that

Reply:
nope, value is string_view, return type cannot be deduced that way

Reply:
I guess not. Result of that gets to **auto&&**, and compiler could not infer type